src/movies.js

```diff
@@ -1,16 +1,118 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(array) {
+    let directorsArr = array.map((movie) => {
+        return movie.director
+    })
+    return directorsArr
+}
+console.log(getAllDirectors(movies));
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array) {
+    let dramaSpielberg = array.filter((movie) => {
+        return (movie.director === "Steven Spielberg") && (movie.genre.includes("Drama"))
+    })
+    return dramaSpielberg.length
+}
+console.log(howManyMovies(movies));
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(array) {
+    if (!array.length) {
+        return 0
+    } else {
+        let allRates = array.reduce((acc, movie) => {
+                if (movie.rate) {
+                    return acc + movie.rate;
+                } else {
+                    return acc + 0
+                }
+            },
+            0);
+        let rateAvg = (allRates / array.length).toFixed(2);
+        rateAvg = Number(rateAvg);
+        return rateAvg
+    }
+}
+console.log(ratesAverage(movies));
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(array) {
```

Comment:
the test not passing here would be solved with the same initial code of iteration 3. 
`if (!array.length) {...`
This happends because along the way the result will become NaN instead of 0 for empty arrays. Using above condition will fix that.